mono/metadata/class.c

```diff
@@ -82,6 +82,74 @@ static void mono_generic_class_setup_parent (MonoClass *klass, MonoClass *gklass
 static gboolean mono_class_set_failure (MonoClass *klass, MonoErrorBoxed *boxed_error);
 
 
+//128 bytes on 64bits
+#define MAX_ADAPT_PAIRS 7
+
+typedef struct {
+	GHashTable *overflow;
+	int count;
+	gpointer keys [MAX_ADAPT_PAIRS];
+	gpointer values [MAX_ADAPT_PAIRS];
+} MonoAdaptatativeHashTable;
+
```

Comment:
Yup, adaptive. Technically they’re synonyms, but adaptative is uncommon/old-fashioned.